docs/output.md

```diff
@@ -223,23 +224,170 @@ print(result.output)
+RouterFailure(explanation="The requested table 'pets' does not exist in the database. The only available table is 'capital_cities', which does not contain data about pets.")
 """
 
 result = router_agent.run_sync('How do I fly from Amsterdam to Mexico City?')
-print(result.output)
+print(repr(result.output))
 """
-explanation = 'I am not equipped to provide travel information, such as flights from Amsterdam to Mexico City.'
+RouterFailure(explanation='I am not equipped to provide travel information, such as flights from Amsterdam to Mexico City.')
 """
 ```
 
+#### Text output
+
+Note that if you provide an output function that takes a string, Pydantic AI will by default create an output tool like for any other output function. If instead you'd like to the model to provide the string using plain text output, you can wrap the function in the [`TextOutput`][pydantic_ai.output.TextOutput] marker class. If desired, this marker class can be used alongside one or more [`ToolOutput`](#tool-output) marker classes (or unmarked types or functions) in a list provided to `output_type`.
+
+```python
+from pydantic_ai import Agent, TextOutput
+
+
+def split_into_words(text: str) -> list[str]:
+    return text.split()
+
+
+agent = Agent(
+    'openai:gpt-4o',
+    output_type=TextOutput(split_into_words),
+)
+result = agent.run_sync('Who was Albert Einstein?')
+print(result.output)
+#> ['Albert', 'Einstein', 'was', 'a', 'German-born', 'theoretical', 'physicist.']
+```
+
+### Output modes
+
+Pydantic AI implements three different methods to get a model to output structured data:
+
+1. [Tool Output](#tool-output)
+2. [Model Structured Output](#model-structured-output)
+3. [Prompted Structured Output](#prompted-structured-output)
+
+#### Tool Output
+
+In the default Tool Output mode, the output JSON schema of each output type (or function) is provided to the model as the parameters schema of a special output tool. This is the default as it's supported by virtually all models and has been shown to work very well.
+
+If you'd like to change the name of the output tool, pass a custom description to aid the model, or turn on or off strict mode, you can wrap the type(s) in the [`ToolOutput`][pydantic_ai.output.ToolOutput] marker class and provide the appropriate arguments. Note that by default, the description is taken from the docstring specified on a Pydantic model or output function, so specifying it using the marker class is typically not necessary.
+
+```python
+from pydantic import BaseModel
+
+from pydantic_ai import Agent, ToolOutput
+
+
+class Fruit(BaseModel):
+    name: str
+    color: str
+
+
+class Vehicle(BaseModel):
+    name: str
+    wheels: int
+
+
+agent = Agent(
+    'openai:gpt-4o',
+    output_type=[
+        ToolOutput(Fruit, name='return_fruit'),
+        ToolOutput(Vehicle, name='return_vehicle'),
+    ],
+)
+result = agent.run_sync('What is a banana?')
+print(repr(result.output))
+#> Fruit(name='banana', color='yellow')
+```
```

Comment:
we're missing lots of `_(This example is complete, it can be run "as is")_` here.

We probably need a linting rule to check them.

But for now just add them.